### PR TITLE
Remove Octave references

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will remove octave from the cmake builds
test the dailes builds more thoroughly
fix all the include what you use warnings from the updated cpplint  (that is most of the changes)

### CODE REVIEW

File: CONTRIBUTORS.md [hunk 1]
```diff
@@ -122,7 +122,7 @@ A networking library with helper functions around Asio and other network interfa
 Several CMake scripts came from other sources and were either used or modified for use in HELICS.
 
 - Lars Bilke [CodeCoverage.cmake](https://github.com/bilke/cmake-modules/blob/master/CodeCoverage.cmake)
-- NATIONAL HEART, LUNG, AND BLOOD INSTITUTE FindOctave.cmake
+- NATIONAL HEART, LUNG, AND BLOOD INSTITUTE
```
Comment (nightlark) [inline CONTRIBUTORS.md:125]: Do we still need this line for another dependency?

File: benchmarks/helics/BenchmarkFederate.hpp [hunk 2]
```diff
@@ -31,49 +31,49 @@ class BenchmarkFederate {
     /** sets the delta time parameter
      * @param dt the delta time to set
```
Comment (nightlark) [inline benchmarks/helics/BenchmarkFederate.hpp:32]: Docstring param name is still the old one

File: benchmarks/helics/BenchmarkFederate.hpp [hunk 3]
```diff
@@ -31,49 +31,49 @@ class BenchmarkFederate {
...
      */
-    void setDeltaTime(helics::Time dt) { deltaTime = dt; }
+    void setDeltaTime(helics::Time delta) { deltaTime = delta; }
     /** gets the delta time parameter*/
     helics::Time getDeltaTime() { return deltaTime; }
     /** sets the final time parameter
      * @param t the final/stop time*/
```
Comment (nightlark) [inline benchmarks/helics/BenchmarkFederate.hpp:38]: Docstring param name is also the old one here

File: src/helics/apps/CMakeLists.txt [hunk 4]
```diff
@@ -91,7 +91,9 @@ if(HELICS_BUILD_APP_LIBRARY)
                                                OR HELICS_DISABLE_ASIO)
     )
         target_compile_definitions(helics_apps PUBLIC BOOST_DATE_TIME_NO_LIB)
-        target_compile_definitions(helics_apps PUBLIC BOOST_BEAST_USE_STD_STRING_VIEW)
+        if(${Boost_VERSION_MINOR} LESS 81)
```
Comment (nightlark) [inline src/helics/apps/CMakeLists.txt:94]: What is Beast using in newer versions of Boost?

File: CONTRIBUTORS.md (hunk 1 above)
Comment (phlptp) [reply CONTRIBUTORS.md:125]: probably not

File: src/helics/apps/CMakeLists.txt (hunk 4 above)
Comment (phlptp) [reply src/helics/apps/CMakeLists.txt:94]: In older versions, we need this flag set to use std::string_view.  In newer versions boost automatically picks the string_view and matches the types so the flag is useless and not needed.

File: interfaces/README.md [hunk 5]
```diff
@@ -1,3 +1,3 @@
 # HELICS Interface extensions (mostly using SWIG)
 
-Currently supports MATLAB, Java, and octave, and C# will generate and is somewhat usable but not completely.
+Currently supports Java, and C# will generate and is somewhat usable but not completely. The other interfaces including python, matlab, and octave in separate repositories.
```
Comment (nightlark) [inline interfaces/README.md:3]:
```suggestion
Currently supports Java, and C# will generate and is somewhat usable but not completely. The other interfaces including Python, Matlab, and Octave are in separate repositories.
```

File: docs/user-guide/installation/language.md [hunk 6]
```diff
@@ -12,33 +12,19 @@ To install HELICS with Java support, you will need to add `HELICS_BUILD_JAVA_INT
...
 
-add the octave folder in the install directory to the octave path
+## HELICS with Octave
 
-```bash
->> helics
->> helicsGetVersion()
-ans = 3.x.x (20XX-XX-XX)
-```
+The Octave interface to HELICS is found at [matHELICS](https://github.com/gmlc-tdc/mathelics) repository. It is currently built with the Matlab Interface
```
Comment (nightlark) [inline docs/user-guide/installation/language.md:23]:
```suggestion
The Octave interface to HELICS is found in the [matHELICS](https://github.com/gmlc-tdc/mathelics) repository. It is currently built with the Matlab Interface.
```

File: docs/user-guide/installation/language.md [hunk 7]
```diff
@@ -12,33 +12,19 @@ To install HELICS with Java support, you will need to add `HELICS_BUILD_JAVA_INT
...
-```bash
-git clone https://github.com/GMLC-TDC/HELICS
-cd HELICS
-mkdir build
-cd build
-cmake -DHELICS_BUILD_OCTAVE_INTERFACE=ON -DCMAKE_INSTALL_PREFIX=/Users/$(whoami)/local/helics-develop/ ..
-make -j8
-make install
-```
+A simulink interface to HELICS is found at [simHELICS](https://github.com/gmlc-tdc/simhelics) repository. This is a minimalistic interface that is build using simulink S functions.
```
Comment (nightlark) [inline docs/user-guide/installation/language.md:19]:
```suggestion
A Simulink interface to HELICS is found in the [simHELICS](https://github.com/gmlc-tdc/simhelics) repository. This is a minimalistic interface that is built using Simulink S-functions.
```

File: docs/user-guide/installation/language.md [hunk 8]
```diff
@@ -12,33 +12,19 @@ To install HELICS with Java support, you will need to add `HELICS_BUILD_JAVA_INT
 
 ## HELICS with MATLAB
 
-The Matlab interface to HELICS is undergoing some major revisions as of HELICS 3.2.1 and is no longer part of the Main HELICS repository. The instructions for use can be found at [matHELICS](https://github.com/gmlc-tdc/mathelics) repository.
+The Matlab interface to HELICS is found at [matHELICS](https://github.com/gmlc-tdc/mathelics) repository.
```
Comment (nightlark) [inline docs/user-guide/installation/language.md:15]:
```suggestion
The Matlab interface to HELICS is found in the [matHELICS](https://github.com/gmlc-tdc/mathelics) repository.
```

**Review by nightlark — APPROVED** [review]

**Pull request merged**